components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -268,137 +261,60 @@ func (w *NotebookWebhook) Handle(ctx context.Context, req admission.Request) adm
-			imgContainer = container
-
-			for key, val := range proxyEnvVars {
-				keyExists := false
-				for _, env := range imgContainer.Env {
-					if key == env.Name {
-						keyExists = true
-						// Update if Proxy spec is updated
-						if env.Value != val {
-							env.Value = val
-						}
-					}
-				}
-				if !keyExists {
-					newVars = append(newVars, corev1.EnvVar{Name: key, Value: val})
-				}
-			}
-
-			// Update container only when required env variables are not present
-			imgContainerExists := false
-			if len(newVars) != 0 {
-				imgContainer.Env = append(imgContainer.Env, newVars...)
-			}
-
-			// Create Volume mount
-			volumeMountExists := false
-			containerVolMounts := &imgContainer.VolumeMounts
-			trustedCAVolMount := corev1.VolumeMount{
-				Name:      "trusted-ca",
+	// Search for the odh-trusted-ca-bundle ConfigMap
+	for i := range configMapList.Items {
+		cm := &configMapList.Items[i]
+		if cm.Name == "odh-trusted-ca-bundle" {
+			fmt.Println("odh-trusted-ca-bundle ConfigMap found")
+
+			volumeName := "trusted-ca"
+			volumeMountPath := "/etc/pki/ca-trust/extracted/pem"
+			volumeMount := corev1.VolumeMount{
+				Name:      volumeName,
+				MountPath: volumeMountPath,
 				ReadOnly:  true,
-				MountPath: "/etc/pki/ca-trust/extracted/pem",
 			}
 
-			for index, volumeMount := range *containerVolMounts {
-				if volumeMount.Name == "trusted-ca" {
-					(*containerVolMounts)[index] = trustedCAVolMount
-					volumeMountExists = true
-					break
-				}
-			}
-			if !volumeMountExists {
-				*containerVolMounts = append(*containerVolMounts, trustedCAVolMount)
+			// Add volume mount to the pod's spec
+			notebook.Spec.Template.Spec.Containers[0].VolumeMounts = append(notebook.Spec.Template.Spec.Containers[0].VolumeMounts, volumeMount)
+
+			// Create volume for mounting the CA certificate from the ConfigMap with key and path
+			configMapVolume := corev1.Volume{
+				Name: volumeName,
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{Name: cm.Name},
+						Optional:             pointer.Bool(true),
+						Items: []corev1.KeyToPath{
+							{
+								Key:  "ca-bundle.crt",
+								Path: "tls-ca-bundle.pem",
+							},
+						},
+					},
+				},
 			}
 
-			// Update container with Env and Volume Mount Changes
-			for index, container := range *notebookContainers {
-				if container.Name == notebook.Name {
-					(*notebookContainers)[index] = imgContainer
-					imgContainerExists = true
-					break
-				}
-			}
+			// Add volume to the pod's spec
+			notebook.Spec.Template.Spec.Volumes = append(notebook.Spec.Template.Spec.Volumes, configMapVolume)
 
-			if !imgContainerExists {
-				return fmt.Errorf("notebook image container not found %v", notebook.Name)
-			}
-			break
+			return nil
 		}
 	}
+	// If odh-trusted-ca-bundle ConfigMap not found
+	fmt.Println("odh-trusted-ca-bundle ConfigMap not found")
```

Comment:
Does the print statement implies something went wrong? If yes, we should return `fmt.error` with this message instead of nil.

Reply:
I will revert that change because maybe the operator is in `unmanaged` mode and  Configmap will not be managed by the operator.